### PR TITLE
Add IBU permissions for ManifestWork agent 

### PR DESCRIPTION
**Pull request by sudomakeinstall2**
ManifestWork agent by default doesn't have permission to access IBU
resource. At the start of IBGU reconcile add the required permissions on the
clusters.
/cc @jc-rh @Missxiaoguo @pixelsoccupied

### CODE REVIEW

File: controllers/imagebasedgroupupgrade.go [hunk 1]
```diff
@@ -142,6 +150,73 @@ func (r *IBGUReconciler) syncStatusWithCGUs(ctx context.Context, ibgu *ibguv1alp
...
+			Namespace: "",
+			Name:      "ibu-permission-set",
+		},
+		Spec: mwv1.ManifestWorkSpec{
+			Workload: mwv1.ManifestsTemplate{
+				Manifests: []mwv1.Manifest{{RawExtension: runtime.RawExtension{Raw: crBytes}}},
+			},
+		},
+	}
+	for _, cluster := range ibgu.Spec.Clusters {
```
Comment (jc-rh) [inline controllers/imagebasedgroupupgrade.go:203]: Is it cleaner to create this as a mwrs and just include it in the CGU?

Comment (sudomakeinstall2) [reply controllers/imagebasedgroupupgrade.go:203]: updated

Comment (jc-rh) [reply controllers/imagebasedgroupupgrade.go:203]: `clusters` actually doesn't include anything if the cluster selector is used. Looks like the only way is to create a mwrs and include it in all CGUs.

Comment (sudomakeinstall2) [reply controllers/imagebasedgroupupgrade.go]: I was thinking about re using the `getclustersforupgrade` which uses both clusters and labelselector.
But I think the mwrs is better.

File: controllers/utils/manifestworkreplicaset.go [hunk 2]
```diff
@@ -121,7 +122,71 @@ func GeneratePrepManifestWorkReplicaset(name, namespace string, ibu *lcav1alpha1
...
+	}
+	if !isUnversioned && len(gvks) == 1 {
+		cr.TypeMeta = metav1.TypeMeta{
+			Kind:       gvks[0].Kind,
+			APIVersion: gvks[0].GroupVersion().Identifier(),
+		}
+	}
+	var dst bytes.Buffer
+	s.Encode(cr, &dst)
+	return dst.Bytes(), nil
```
Comment (pixelsoccupied) [inline controllers/utils/manifestworkreplicaset.go:186]: Ah surprised that upstream doesn't provided a simple byte conversion func. Seems like super useful use case. Nothing in the Unstructured Objects lib either yeah?

**Review by pixelsoccupied — COMMENTED** [review]
Just a few comments but lgtm otherwsie!

File: controllers/utils/manifestworkreplicaset.go (hunk 2 above)
Comment (pixelsoccupied) [reply controllers/utils/manifestworkreplicaset.go:186]: How about something like this https://github.com/open-cluster-management-io/api/blob/bd6f2229b53cb61c76ebf528c73bc925d8e0f4c8/utils/work/v1/workapplier/workapplier.go#L166-L181 ?

File: controllers/utils/manifestworkreplicaset.go [hunk 3]
```diff
@@ -59,8 +60,8 @@ func GenerateFinalizeManifestWorkReplicaset(name, namespace string, ibu *lcav1al
 }
 
 // GenerateUpgradeManifestWorkReplicaset returns a populated ManifestWorkReplicaSet for upgrade stage of an ImageBasedUpgrade
-func GenerateUpgradeManifestWorkReplicaset(name, namespace string, ibu *lcav1alpha1.ImageBasedUpgrade) (*mwv1alpha1.ManifestWorkReplicaSet, error) {
-	ibu.Spec.Stage = lcav1alpha1.Stages.Upgrade
```
[the remark this answers is not in the document]
Comment (sudomakeinstall2) [reply controllers/utils/manifestworkreplicaset.go:63]: sure, https://github.com/openshift-kni/cluster-group-upgrades-operator/pull/789

File: controllers/utils/manifestworkreplicaset.go (hunk 2 above)
Comment (sudomakeinstall2) [reply controllers/utils/manifestworkreplicaset.go:186]: this does not work. It doesn't include apiversion, group and kind.

Comment (pixelsoccupied) [reply controllers/utils/manifestworkreplicaset.go:186]: Good to know! Thanks for checking

**Review by jc-rh — APPROVED** [review]

**Pull request merged**